### ISSUE TITLE
[REG] Failed to find devices

### ISSUE THREAD

**Issue by wanghongjuan** [siovene/cordova-plugin-oic-demo#11]
## Test Environments:
- HW: Edison, Mini Fan v1.0
- SW: Ostro image [build#481](https://download.ostroproject.org/builds/ostro-os/2016-07-19_23-56-09-build-481/images/edison/ostro-image-swupd-dev-edison-2016-07-19_23-56-09-build-481.toflash.tar.bz2), OIC resource server: [fan.js](https://github.com/01org/SmartHome-Demo/blob/master/ocf-servers/js-servers/fan.js)
## Steps to reproduce:
1. Following the file [README](https://github.com/01org/SmartHome-Demo/blob/master/ocf-servers/js-servers/README.md), connect the fan sensor to Edison, and start the sensor.
2. Launch this demo, press **Settings** in the menu list and enable "Continuous discovery"
3. Switch to **Resources**, show the fan sensor's information. Then switch to "Devices",  show nothing. But it will show the devices information when run this demo using commit `792e64`

**Comment by ghost** [comment]
I see no such issue in my setup using version 0.6.

**Comment by wanghongjuan** [comment]
Hi, @uhofemei could you please tell me which commit you are using? Since I can't find the version 0.6, this demo just has one release version v0.1.0.

**Comment by ttzeng** [comment]
In the later Ostro images, most ports are blocked by default. Try "iptables -A INPUT -p udp --dport 5683 -j ACCEPT" to pass through the CoAP packets.

**Comment by wanghongjuan** [comment]
> > In the later Ostro images, most ports are blocked by default. Try "iptables -A INPUT -p udp --dport 5683 -j ACCEPT" to pass through the CoAP packets.

@ttzeng After setting the port `5683`, it still doesn't work.

**Comment by ttzeng** [comment]
@wanghongjuan would you list your firewall rules by "iptables -S"?

**Comment by wanghongjuan** [comment]
Hi, @ttzeng  The firewall rules information show as below:

```
root@edison:~# iptables -S              
-P INPUT DROP
-P FORWARD DROP
-P OUTPUT ACCEPT
-A INPUT -p tcp -m tcp --dport 2000 -j ACCEPT
-A INPUT -i ve-+ -p udp -m udp --dport 53 -j ACCEPT
-A INPUT -i ve-+ -p udp -m udp --dport 67 -j ACCEPT
-A INPUT -i lo -j ACCEPT
-A INPUT -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-A INPUT -p tcp -m tcp --dport 22 -j ACCEPT
-A INPUT -p udp -m udp --dport 5683 -j ACCEPT
-A FORWARD -o ve-app-4 -j ACCEPT
-A FORWARD -o ve-+ -j ACCEPT
-A FORWARD -i ve-+ -j ACCEPT
```

**Comment by ttzeng** [comment]
@wanghongjuan, well, the kernel can pass through CoAP packets, from the description, you were running fan.js. could you enable the debug message (export NODE_DEBUG=fan) and see any errors? If the server registers the resource successfully, and the kernel unblock the CoAP traffic, it should receive requests from the clients.

**Comment by wanghongjuan** [comment]
@ttzeng The fan server doesn't show any errors after enable the debug message and shows the server register the resource successfully, but this issue is still exists. Actually, if using latest Ostro image, the resources and the devices information can not be found, but if using Ostro image#481, the resources can be found. Forward comment from @nagineni, the iotivity doesn't work, because there are some changes of firewall in latest Ostro image. I'm not sure whether it is caused by the firewall setting you mentioned.   I'm not clear about the changed setting.

**Comment by nagineni** [comment]
@wanghongjuan It's related to the tight firewall settings in the Ostro. Create holes to the firewall like in the commit https://github.com/ostroproject/meta-iotqa/pull/194/commits/485a838 to make the discovery works.

**Comment by ttzeng** [comment]
@wanghongjuan though I never tried Ostro #481, but #497 works for me after unblock the 5683 port on UDP stream. What image build # exactly you failed to discover the fan server? I notices #504 just posted yesterday, I was assuming you were fail with #497?

**Comment by wanghongjuan** [comment]
@nagineni Thanks for your confirmation and information, they are make sense to me. @ttzeng I tried #497 before the #504 was posted. Yesterday I tried to flash the image#504 on Edison board and re-run this demo, show results as below:

|  | Before Setting Firewall (#504) | After Setting Firewall (#504) | Default (#481) |
| --- | --- | --- | --- |
| Discovery Devices(commit [537471c](https://github.com/siovene/cordova-plugin-ocf-demo/commit/537471c2165c38258988f2acfee496cd0afa3e59)) | FAIL | FAIL | FAIL |
| Discovery Resources(commit [537471c](https://github.com/siovene/cordova-plugin-ocf-demo/commit/537471c2165c38258988f2acfee496cd0afa3e59)) | FAIL | PASS | PASS |
| Discovery Devices(commit [792e64e](https://github.com/siovene/cordova-plugin-ocf-demo/commit/792e64e6f380028960367bf8ff0465074205cc1e)) | FAIL | PASS | PASS |
| Discovery Resources(commit [792e64e](https://github.com/siovene/cordova-plugin-ocf-demo/commit/792e64e6f380028960367bf8ff0465074205cc1e)) | FAIL | PASS | PASS |

Good news is #504 is works after setting the firewall and #497 has same result as #504.
BTW, the devices can be found by http request when iot-rest-api-server is start, the http request command is `curl http://[device-ip]:8000/api/oic/d`, then return json format information:

```
[{"di":"39666334-6433-3933-2d65-3433312d3466","n":"Smart Home Fan","icv":"core.1.0.0"},{"di":"39666334-6433-3933-2d65-3433312d3466","n":"Smart Home Fan","icv":"core.1.0.0"}]
```

So I think this issue is caused by different commit, it doesn't matter with Ostro image. How do you think about this?

**Comment by wanghongjuan** [comment]
@siovene Could you please take a look?

**Comment by siovene** [comment]
I'm investigating.

**Comment by siovene** [comment]
This is now fixed in `cordova-ocf-plugin` v0.6.3. To test the fix, you don't need to upgrade this package (tho it's recommended because I did some updates here too), but only need to make sure that you use the correct version of `cordova-ocf-plugin`:

```
cordova plugin rm cordova-ocf-plugin
cordova plugin add cordova-ocf-plugin
```

(make sure that v0.6.3 is fetched from `npm` .

**Comment by qiuzhong** [comment]
Verified. OCF devices can be disovered via the cordova-plugin-ocf-demo client built on cordova-ocf-plugin v0.6.3.
